crates/primitives/src/block.rs

```diff
@@ -467,6 +467,7 @@ impl<H, B> reth_primitives_traits::Block for SealedBlock<H, B>
 where
     H: reth_primitives_traits::BlockHeader + 'static,
     B: reth_primitives_traits::BlockBody + 'static,
+    SealedHeader<H>: MaybeArbitrary,
```

Comment:
instead of this bound here, relax the arbitrary impl for `SealedHeader`
```rust
#[cfg(any(test, feature = "arbitrary"))]
impl<'a, H> arbitrary::Arbitrary<'a> for SealedHeader<H> where H: for<'a> arbitrary::Arbitrary<'a> {
    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
        let header = Header::arbitrary(u)?;

        Ok(Self::seal(header))
    }
}
```